src/modules/Instabug.ts

```diff
@@ -56,44 +73,198 @@ function reportCurrentViewForAndroid(screenName: string | null) {
+    const isUpdated = setApmNetworkFlagsIfChanged(updatedFlags);
+
+    if (isUpdated) {
+      console.log('Andrew: APM network flags updated.');
+      handleNetworkInterceptionMode(config);
+      initializeNativeInstabug(config);
+    }
+    _logFlags();
+    console.log('Andrew: App has come to the foreground!');
+  }
+
+  _currentAppState = nextAppState;
+  console.log(`Andrew: Current AppState: ${_currentAppState}`);
+};
+
+/**
+ * Fetches the current APM network flags.
+ */
+const fetchApmNetworkFlags = async () => {
+  isNativeInterceptionFeatureEnabled = await NativeNetworkLogger.isNativeInterceptionEnabled();
+  if (Platform.OS === 'android') {
+    hasAPMNetworkPlugin = await NativeNetworkLogger.hasAPMNetworkPlugin();
+  }
+
+  return {
+    isNativeInterceptionFeatureEnabled,
+    hasAPMNetworkPlugin,
+    shouldEnableNativeInterception,
+  };
+};
+
+/**
+ * Handles platform-specific checks and updates the network interception mode.
+ */
+const handleNetworkInterceptionMode = (config: InstabugConfig) => {
+  // Default networkInterceptionMode to JavaScript if not set
   if (config.networkInterceptionMode == null) {
     config.networkInterceptionMode = NetworkInterceptionMode.javascript;
   }
 
+  if (Platform.OS === 'android') {
+    checkNativeInterceptionForAndroid(config);
+    config.networkInterceptionMode = NetworkInterceptionMode.javascript; // Need to enable JS interceptor in all scenarios for Bugs & Crashes network logs
+  } else if (Platform.OS === 'ios') {
+    checkNativeInterceptionForIOS(config);
+  }
+
   if (config.networkInterceptionMode === NetworkInterceptionMode.javascript) {
     NetworkLogger.setEnabled(true);
   }
+};
+
+/**
+ * Handles the native interception logic for Android.
+ */
+const checkNativeInterceptionForAndroid = (config: InstabugConfig) => {
+  const { networkInterceptionMode } = config;
+
+  if (networkInterceptionMode === NetworkInterceptionMode.javascript) {
+    if (isNativeInterceptionFeatureEnabled && hasAPMNetworkPlugin) {
+      shouldEnableNativeInterception = true;
+      console.warn(
+        InstabugConstants.IBG_APM_TAG + InstabugConstants.SWITCHED_TO_NATIVE_INTERCEPTION_MESSAGE,
+      );
+    }
+  } else {
+    if (isNativeInterceptionFeatureEnabled) {
+      shouldEnableNativeInterception = hasAPMNetworkPlugin;
+      if (!hasAPMNetworkPlugin) {
+        console.error(
+          InstabugConstants.IBG_APM_TAG + InstabugConstants.PLUGIN_NOT_INSTALLED_MESSAGE,
+        );
+      }
+    } else {
+      if (hasAPMNetworkPlugin) {
+        console.error(
+          InstabugConstants.IBG_APM_TAG + InstabugConstants.NATIVE_INTERCEPTION_DISABLED_MESSAGE,
+        );
+      } else {
+        shouldEnableNativeInterception = false; // rollback to use JS interceptor for APM & Core.
+        console.error(
+          InstabugConstants.IBG_APM_TAG +
+            InstabugConstants.PLUGIN_NOT_INSTALLED_AND_NATIVE_INTERCEPTION_DISABLED_MESSAGE,
+        );
+      }
+    }
+  }
+};
```

Comment:
I have extracted several helper methods from this class, including `handleAppStateChange`, `setApmNetworkFlagsIfChanged`, `checkNativeInterceptionForAndroid`, and `checkNativeInterceptionForIOS`, among others. I aimed to avoid excessive splitting of the code to maintain readability. Additionally, the well-named variables and use-case comments help clarify the scenarios, which are not numerous.

@kholood-ea what do you think?

Reply:
its recommended that each function does only one thing, multiple nested if else's might give an indication that it doesn't. 

you already mentioned well named variables, I think that along with well code structuring such as yours wont make me worry too much about code splitting.